lesson14/ht_template.ipynb

```diff
@@ -0,0 +1,420 @@
+     "output_type": "stream",
+     "text": [
+      "Setting default log level to \"WARN\".\n",
+      "To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "22/10/22 17:23:55 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable\n"
+     ]
+    }
+   ],
+   "source": [
+    "spark = SparkSession.builder.appName('lect_13_home_task').getOrCreate()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "# Домашнє завдання на тему Spark SQL\n",
+    "\n",
+    "Задачі з домашнього завдання на SQL потрібно розвʼязати за допомогою Spark SQL DataFrame API.\n",
+    "\n",
+    "Дампи таблиць знаходяться в папці `data`.\n",
+    "Можете створювати стільки нових клітинок, скільки вам необхідно.\n",
+    "\n",
+    "Розвʼязок кожної задачі має бути відображений в самому файлі (використати метод `.show()`)\n",
+    "\n",
+    "**Увага!** Використовувати мову запитів SQL безпосередньо забороняється, потрібно використовувати виключно DataFrame API!\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "#### 1. Вивести кількість фільмів в кожній категорії. Результат відсортувати за спаданням."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "+-----------+-----+\n",
+      "|category_id|count|\n",
+      "+-----------+-----+\n",
+      "|         15|   74|\n",
+      "|          9|   73|\n",
+      "|          8|   69|\n",
+      "|          6|   68|\n",
+      "|          2|   66|\n",
+      "|          1|   64|\n",
+      "|         13|   63|\n",
+      "|          7|   62|\n",
+      "|         10|   61|\n",
+      "|         14|   61|\n",
+      "|          3|   60|\n",
+      "|          5|   58|\n",
+      "|         16|   57|\n",
+      "|          4|   57|\n",
+      "|         11|   56|\n",
+      "|         12|   51|\n",
+      "+-----------+-----+\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "df = spark.read.options(header='True', inferSchema='True').csv(\"data/film_category.csv\")\n",
+    "df.groupBy('category_id').count().sort('count', ascending=False).show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "#### 2. Вивести 10 акторів, чиї фільми брали на прокат найбільше. Результат відсортувати за спаданням."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 53,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "+--------+\n",
```

Comment:
I expect to see the actor's name here. No real sense in viewing ids, right?

Reply:
You are totally right. However, time pressed... So, I did this as quick as possible.